AWSPinpointTests/AWSPinpointTargetingClientTests.m

```diff
@@ -55,10 +55,11 @@ - (void)setUp {
     [[NSUserDefaults standardUserDefaults] removeSuiteNamed:@"AWSPinpointTargetingClientTests"];
     self.userDefaults = [[NSUserDefaults alloc] initWithSuiteName:@"AWSPinpointTargetingClientTests"];
 
-    [self initializeMockApplicationWithOptOut:YES];
+    [self initializeMockApplicationWithRemoteNotifications:NO];
     [self initializePinpointWithConfiguration:[self getDefaultAWSPinpointConfiguration] forceCreate:NO];
```

Comment:
We should remove these lines here, and put them in individual functions (so that we aren't reinitializing state)

Reply:
that one should be covered by line 250, `testCurrentProfileReturnsOptOutAllWhenNotificationsEnabledAndDeviceTokenNotSet` and this test based on executing on background thread, i think, was meant mainly to test calling into the main thread with remote notifications enabled. but doesn't hurt add more test coverage!

Reply:
i agree, it's hard to know what's going on when the setUp portion of the code is setting up the arrange part of a test, especially the mock application with remote notifications set to false